src/pybind/mgr/dashboard/frontend/src/app/shared/services/summary.service.spec.ts

```diff
@@ -66,14 +67,80 @@ describe('SummaryService', () => {
     subs.unsubscribe();
   }));
 
+  describe('Should test subscribe without initial value', () => {
+    beforeEach(() => {
+      summaryService['summaryDataSource'].next(undefined);
+    });
+
+    it('should call subscribeOnce', () => {
+      let result;
+      let i = 0;
+      const subscriber = summaryService.subscribeOnce((response) => {
+        i++;
+        result = response;
+      });
+
+      expect(subscriber).toEqual(jasmine.any(Subscriber));
+      expect(i).toBe(0);
+      expect(result).toEqual(undefined);
+
+      summaryService['summaryDataSource'].next(undefined);
+      expect(i).toBe(0);
+      expect(result).toEqual(undefined);
+      expect(subscriber.closed).toBe(false);
+
+      summaryService['summaryDataSource'].next(summary);
+      expect(result).toEqual(summary);
+      expect(i).toBe(1);
+      expect(subscriber.closed).toBe(true);
+
+      summaryService['summaryDataSource'].next(summary);
+      expect(result).toEqual(summary);
+      expect(i).toBe(1);
+    });
+
+    it('should call subscribe', () => {
+      let result;
+      let i = 0;
+      const subscriber = summaryService.subscribe((data) => {
+        i++;
+        result = data;
+      });
+
+      expect(subscriber).toEqual(jasmine.any(Subscriber));
+      expect(i).toBe(0);
+      expect(result).toEqual(undefined);
+
+      summaryService['summaryDataSource'].next(undefined);
+      expect(i).toBe(0);
+      expect(result).toEqual(undefined);
+      expect(subscriber.closed).toBe(false);
+
+      summaryService['summaryDataSource'].next(summary);
+      expect(result).toEqual(summary);
+      expect(i).toBe(1);
+      expect(subscriber.closed).toBe(false);
+
+      summaryService['summaryDataSource'].next(summary);
+      expect(result).toEqual(summary);
+      expect(i).toBe(2);
+      expect(subscriber.closed).toBe(false);
+    });
+  });
+
   describe('Should test methods after first refresh', () => {
     beforeEach(() => {
       authStorageService.set('foobar', undefined, undefined);
       summaryService.refresh();
     });
 
-    it('should call getCurrentSummary', () => {
-      expect(summaryService.getCurrentSummary()).toEqual(summary);
+    it('should call subscribeOnce', () => {
```

Comment:
It looks like an incomplete duplicate of the test above with the same name.

Reply:
Will remove